### ISSUE TITLE
Setting `track_running_stats=False` for normalization of embeddings

### ISSUE THREAD

**Issue by adosar** [zcao0420/MOFormer#12]
First of all, I would like to congratulate you on your excellent work and for open sourcing the implementation 👏!

I have a question regarding the calculation of BarlowTwins loss. Embeddings are normalized using batch normalization without setting `track_running_stats=False`, meaning that `mean` and `std` for inference (i.e. when setting `model.eval()`) are based on a moving average from the training batch statistics. Do you think it would be more appropriate to initialize `self.bn` with `track_running_stats=False` so loss is calculate in same way both during training and inference (validation/test)?

```python
class BarlowTwinsLoss(nn.Module):
    def __init__(self,device,batch_size,embed_size,lambd):
        super(BarlowTwinsLoss,self).__init__()
        self.device = device
        self.embed_size = embed_size
        self.batch_size = batch_size
        self.lambd = lambd # default=0.005
        self.bn = nn.BatchNorm1d(self.embed_size, affine=False).to(self.device)  # Add track_running_stats=False.

    def forward(self, z1, z2):

        # empirical cross-correlation matrix
        c = self.bn(z1).T @ self.bn(z2).to(self.device)

        # sum the cross-correlation matrix between all gpus
        c.div_(self.batch_size)
        #torch.distributed.all_reduce(c)

        on_diag = torch.diagonal(c).add_(-1).pow_(2).sum().to(self.device)
        off_diag = off_diagonal(c).pow_(2).sum().to(self.device)
        loss = on_diag + self.lambd * off_diag
        return loss.to(self.device)
```

**Comment by RishikeshMagar** [comment]
Hi @adosar,
Thanks for the question, I believe that was an oversight on our end. Your question makes sense and the model may show better inference performance indeed. Unfortunately, we dont have the compute resources to retrain and test this.

Another way to look at it is because we did the random split during training and validation, it is highly likely that the running `mean` and `std` were similar for both train and validation. Of course, we can't show this now but this is also a possibility
